guides/database-integrating-with-postgresql.adoc

```diff
@@ -0,0 +1,739 @@
+
+In addition to the Galleon Layers to configure the datasource and install the drivers, we also want to trim the WildFly server to remove any unnecessary subsystems and features we don't need. That will reduce the server footprint and the security attack surface. This task can be done by selecting the appropriate https://docs.wildfly.org/32/Galleon_Guide.html#wildfly_galleon_layers[Galleon Layers, window=_blank] shipped with any WildFly distribution. However, instead of adding a static list of Galleon Layers, we are going to configure the `wildfly-maven-plugin` plugin to discover the required layers automatically for us.
+
+Replace the current `wildfly-maven-plugin` configuration in the `pom.xml` file provided by the getting started guide with the following one:
+
+[source,xml]
+----
+<plugin>
+    <groupId>org.wildfly.plugins</groupId>
+    <artifactId>wildfly-maven-plugin</artifactId>
+    <version>${version.wildfly.maven.plugin}</version>
+    <configuration>
+        <discover-provisioning-info>
+            <addOns>
+                <addOn>postgresql:default</addOn>
+            </addOns>
+        </discover-provisioning-info>
+    </configuration>
+    <executions>
+        <execution>
+            <goals>
+                <goal>package</goal>
+            </goals>
+        </execution>
+    </executions>
+</plugin>
+----
+In the above configuration, behind the scenes the `wildfly-maven-plugin` is using https://docs.wildfly.org/wildfly-glow/[WildFly Glow, window=_blank] to discover automatically the required Galleon Layers for our application. The `discover-provisioning-info` configuration tells the plugin to discover the required layers by inspecting our application code. By using the `postgresql:default` addon, we are specifying we want to use a PostgreSQL database, and we want to configure it as the default datasource for the server.
+
+=== persistence.xml
+This file is used to configure the Jakarta Persistence API (JPA) persistence unit and its database schema generation strategy. In this guide, we are using the `drop-and-create` strategy to drop the existing schema and create a new one every time the application starts. For a production environment, you should use a more appropriate strategy to avoid data loss.
```

Comment:
Since we're dealing with the abbreviations, 'JPA' is another one we should drop.

Reply:
Also, no "API".

It's just Jakarta Persistence.  https://jakarta.ee/specifications/persistence/

Reply:
Done. It leads to "This file is used to configure the Jakarta Persistence persistence unit", which indeed seems to be correct.